### ISSUE TITLE
PHP Parse error:  syntax error, unexpected 'class' (T_CLASS) in param-processor/DefaultConfig.php on line 24

### ISSUE THREAD

**Issue by DamienGombaultRecia** [SemanticMediaWiki/SemanticMediaWiki#2015]
### Setup and configuration

- SMW version: 2.4.2
- MW version: 1.26.4
- PHP version: 5.4.16 (CentOS 7.2, up to date)
- DB (MySQL etc.): 5.5.50-MariaDB (CentOS 7.2, up to date)

### Issue

Hi.
I have upgraded my MediaWiki installation to version 1.26.4 with Semantic MediaWiki 2.4.2.
When I run the maintenance/update.php MediaWiki PHP script, I get this error :
```
PHP Parse error:  syntax error, unexpected 'class' (T_CLASS), expecting identifier (T_STRING) or variable (T_VARIABLE) or '{' or '$' in [...]/extensions/SemanticMediaWiki/vendor/param-processor/param-processor/DefaultConfig.php on line 24
```

I have downgraded to version 2.4.1, it works.

**Comment by kghbln** [comment]
@DamienGombaultRecia Thanks for reporting!
@JeroenDeDauw ParamProcessor is your speciality I guess.

**Comment by DamienGombaultRecia** [comment]
I think this is caused by :
ParamProcessor
1.3.0 (2016-07-15)
    Dropped support for PHP 5.3 and PHP 5.4.

I still use PHP 5.4.

SMW 2.4.x supports PHP 5.3.2 - 7.0 (from https://github.com/SemanticMediaWiki/SemanticMediaWiki/blob/master/docs/COMPATIBILITY.md#platform-compatibility), but included dependencies does not support PHP 5.3 or 5.4 anymore.

What is the right solution ?
Create a bundle release with old dependencies release (ParamProcessos 1.2.5) which still support PHP 5.3 or 5.4 or upgrade minimal requirement to PHP 5.5 for SMW 2.4.x ?

**Comment by kghbln** [comment]
> I still use PHP 5.4.

Ah, good find. Great.

> upgrade minimal requirement to PHP 5.5

You will have to do this sooner than later anyway since PHP 5.4 is unsupported in upstream.

Still I think that SWM should only fetch ParamPrecessor matching overall PHP compat requirments. Let's see what @JeroenDeDauw

**Comment by JeroenDeDauw** [comment]
@DamienGombaultRecia you got SMW via the tarball release or what? That indeed is not holding into account that the PHP version of the target machine might be older than that of the build machine... I have filed this now as https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/2016

> I have downgraded to version 2.4.1, it works.

Uh... I wonder why that'd be different. Perhaps @kghbln build the 2.4.1 tarballs on a box with PHP 5.4 or PHP 5.3? Or perhaps the oversight in the tarball build is not the issue, and we backported PHP 5.5 stuff onto 2.4.x for 2.4.2...

**Comment by JeroenDeDauw** [comment]
@DamienGombaultRecia but yeah, upgrading to 5.6 or 7.0 is recommend anyway

**Comment by kghbln** [comment]
> Uh... I wonder why that'd be different. Perhaps @kghbln build the 2.4.1 tarballs on a box with PHP 5.4

If tarballs a used this could indeed be a reason since in July my box was still on Debian 7 using PHP 5.4

**Comment by nakohdo** [comment]
I got a similar parse error while trying a tarball upgrade from SMW 1.8.0.5 to 2.4.3 with the following setup. Upgrading to 2.4.1 - as suggested above - did work.

- MediaWiki 1.19.23
- PHP 5.3.3 (apache2handler)
- MySQL 5.1.73

Error message:
> PHP Parse error: unexpected '[' in param-processor/DefaultConfig.php on line 18.

**Comment by mwjames** [comment]
@kghbln @JeroenDeDauw In case you make a 2.4.4 tag then I hope we can get rid of this issue!

**Comment by kghbln** [comment]
> @kghbln @JeroenDeDauw In case you make a 2.4.4 tag then I hope we can get rid of this issue!

As soon as https://github.com/SemanticMediaWiki/SemanticMediaWiki/issues/2016 is fixed this one is fixed, too. Yeah, this will be great for 2.4.4. @JeroenDeDauw Any idea how this can be done.

**Comment by kghbln** [comment]
@nakohdo Could you try [SMW 2.4.4](https://github.com/SemanticMediaWiki/SemanticMediaWiki/releases/tag/2.4.4) and see if the tarball release works for you now.

**Comment by nakohdo** [comment]
@kghbln Will try if possible (this is a customer site where I don't have direct server access).

**Comment by kghbln** [comment]
Ah, I get it. I see that @DamienGombaultRecia could try too. However is first. :)

**Comment by DamienGombaultRecia** [comment]
I have upgraded my installation to PHP 5.6 as recommended.

**Comment by kghbln** [comment]
> I have upgraded my installation to PHP 5.6 as recommended.

That's even better!

**Comment by nakohdo** [comment]
Have tried a tarball upgrade with https://github.com/SemanticMediaWiki/SemanticMediaWiki/releases/download/2.4.4/Semantic.MediaWiki.2.4.4.dependencies.tgz but still get the same PHP error:
 
``PHP Parse error:  syntax error, unexpected '[' in /extensions/SemanticMediaWiki/vendor/param-processor/param-processor/DefaultConfig.php on line 18``

But we are happy to stay with 2.4.1 until the PHP upgrade, considering that we were still using 1.7.0.2 two weeks ago ;-)

**Comment by kghbln** [comment]
@nakohdo Thanks a lot for testing though the result is still desagreable. Well 2.4.1 is pretty good and all subsequent release mostly addressed environments that you do not yet use.

**Comment by kghbln** [comment]
I am being bold here by adding the 2.4.5 milestone. Touching wood.

**Comment by kghbln** [comment]
Releasable to the 2.5.0 milestone. It is a bigger issue than expected to provide tarballs for antediluvian setups. :(

**Comment by mwjames** [comment]
> Re-label to the 2.5.0 milestone

The syntax used by `param-processor/param-processor/DefaultConfig.php` requires PHP 5.5 and now that 2.5 will be released soon, PHP 5.5 is a prerequisite for SMW as well therefore closing this issue.

**Comment by kghbln** [comment]
Solved by time.
